backend/lib/til_web/controllers/post_controller.ex

```diff
@@ -4,13 +4,22 @@ defmodule TilWeb.PostController do
   alias Til.ShareableContent
   alias Til.Notifications
 
-  def index(conn, params) do
+  def index(conn, %{"q" => _} = params) do
```

Comment:
I don't think that we need to guard it here, it can we handled deeper with transparent interface